src/gui/src/scriptWidget.cpp

```diff
@@ -385,8 +412,15 @@ class ScriptWidget::GuiSink : public spdlog::sinks::base_sink<Mutex>
         std::string(formatted.data(), formatted.size()));
 
     if (msg.level == spdlog::level::level_enum::off) {
-      // this comes from a ->report
-      widget_->addReportToOutput(formatted_msg);
+      if (!widget_->reportTimerIsActive()) {
+        widget_->startReportTimer();
```

Comment:
this can only be started from the main thread, you need to guard against this. otherwise you will get a bunch of errors about this timer.